### DISCUSSION TITLE
Altitude of the single layer mode

### DISCUSSION THREAD

**Discussion by FFFFFG-IGG** [groops-devs/groops#240] [Q&A]
Hello developers! Generally, altitude of single layer mode (SLM) ranges from 350km to 450km. But I found that the default layer altitude in Groops is 506.7Km (See "IonosphereVTEC"), which makes me suspicious. With this in mind, I checked the paper "Processing of GNSS constellations and ground station networks using the raw observation approach (Section 3.3)" and Sebastian's PhD thesis (Section 6.4.1 and Section 6.4.2), and found they used different altitude of SLM (350km, 450km, 506.7km, respectively). This make me suspicious on two points: **1. Does the altitude 506.7km a mistake or a improvement (modification) in the up-to-dated paper (If is, could you please give a link)?  2. The altitude of SLM in 6.4. and 6.4.2 for Sebastian's PhD is different. When demonstrating the second order term of Ionospheric correction, it claims that layer altitude is 450km. However, when it comes to mapping function in following section 6.4.2, the layer altitude changes to 506.7km. Could you give me some consults about this differences?** 3. With above issues in mind, I'm also interested in how to obtain the lontitude and latitude of of the pierce point of SLM in your strategy?
I would aprreciate for you patience and look forward to your kind reply!
kind regards, 
Y Fan.

**Comment by tmayerguerr** [comment]
Hi Y Fan,

The ionosphere is still an open issue in groops. We hope that we can continue to work on this topic this year.
 
The values are taken from the Bernese GNSS Software User Handbook:
>  The “modified” SLM (MSLM) mapping function includes an additional constant, alpha (Schaer 1999): Best fit of Eqn. (13.8) with respect to the JPL extended slab model (ESM) mapping function is achieved at H = 506.7 km and alpha = 0.9782 (when using R = 6371 km and assuming a maximum zenith distance of 80 degrees). The resulting mapping function is used in the ionosphere analysis at CODE. For computation of the ionospheric pierce points, H = 450 km is assumed for ground stations.

Unfortunately, we have overlooked the fact that the height H of the mapping function and the height of the pierce point are different. At the moment, mapH is used for both in GROOPS. This could explain why our ionospheric estimation gives strange results. Many thanks for the hint.  I'll see if I can find a temporary solution, but at the moment we have a lot of other things to do.

Best regards
Torsten

**Reply by FFFFFG-IGG** [reply]
Thanks for your reply. You mean groops use 507.6km in both pierce point and mapping function calculation? Due to this, the Ionosphere delay is strange? May be changing the pierce point height in Ionosphere delay parameter setting with 450km could help.

**Reply by FFFFFG-IGG** [reply]
Hi,
I‘ve checked the source code and find that height of pierce point is 450km when processing GNSS satellites, while the height is slightly higher (50km above the satellite). If I were right, these setteings in groops only lead to some strange VTEC values of LEO-based but not harm to GNSS-based satellites (whose height is much more higher).  Hope to get some kind consults from your guys!
Kind regards 
Y Fan
